src/mlagility/analysis/analysis.py

```diff
@@ -438,22 +455,50 @@ def forward_spy(*args, **kwargs):
                     depth,
                     parent_hash,
                 )
+
+            # Get parent workload hash
+            parent_workload_hash = None
+            if parent_hash:
+                parent_workload_hash = tracer_args.models_found[
+                    parent_hash
+                ].last_workload_executed
+
             model_hash = get_model_hash(local_var, model_type)
+            workload_hash, input_shapes = get_workload_hash(
+                model_hash, parent_workload_hash, args, kwargs
+            )
             model_info = tracer_args.models_found[model_hash]
-            model_info.exec_time = model_info.exec_time + end_time - start_time
 
-            model_info.executed = model_info.executed + 1
+            if workload_hash not in model_info.workloads:
+                model_info.workloads[workload_hash] = util.WorkloadInfo(
+                    hash=workload_hash,
+                    is_target=workload_hash in tracer_args.targets
+                    or tracer_args.targets == [],
```

Comment:
```suggestion
                    or len(tracer_args.targets) == 0,
```

nit: I'm not 100% sure, but would it be more explicit/good to test this instead? Since that gets to the heart of what you really care about (there are no specific targets).

Seems good to me because if we changed `tracer_args.targets` to a dict in the future, no crazy shit would happen here.

Reply:
Changed as recommended!